### PR TITLE
Back to the future (0.89 -> 0.95)

### PR DESCRIPTION
**Pull request by DominiqueMakowski**
Yes yes it's finally there and y'all gonna hate me. But you gotta do what you gotta do, even if it leads to tomatoes being thrown at you 🙃 

<sub>*And I can't stand this message on loading any longer* 😬</sub>

### CODE REVIEW

**Comment by mattansb** [comment]
I still am not sure why we're doing this?

Make sure to get all 100,000,000 places where it's 0.89...

---

Also, why aren't we using 
```R
options(bayestestR_ci = 0.89) # on startup


hdi <- function(x, ci = getOption("bayestestR_ci")) {
  ...
}
```

**Comment by DominiqueMakowski** [comment]
> Make sure to get all 100,000,000 places where it's 0.89...

Turns out I made a search and it's only few... like really a handful

**Comment by DominiqueMakowski** [comment]
> Also, why aren't we using

Coz that's even more occult to new users, at least if we write it in the function declaration it's clear and obvious

File: R/zzz.R [hunk 1]
```diff
@@ -1,5 +1,8 @@
 .onAttach <- function(libname, pkgname) {
+  message <- "bayestestR is using a not-recommended default-level for uncertainty intervals. Note that you probably need an effective sample size of at least 10.000 per parameter to get robust tails of your uncertainty intervals. Unfortunately, the default number of iterations for most Bayes packages return only 4.000 samples of the posterior distribution. Be prepared to set a sensible default number of interation during model ftting, or set the 'ci'-argument in bayestestR function explicitly to a value of max. 0.9."
```
Comment (DominiqueMakowski) [inline R/zzz.R:2]: my god Daniel ^^

Comment (mattansb) [reply R/zzz.R:2]: Yessssssss BOSS MOVE!

**Comment by DominiqueMakowski** [comment]
Here's finally some preliminary home made data related to this infamous STABILITY (I don't have time to run more samples as I need to go sleep but I count on you to let it run 'til 10.000 with more iterations per draws so that I can marvel at the new plot when I wake up ☺️ )

image


```r
library(dplyr)
library(ggplot2)
library(rstanarm)

data <- data.frame()
for(draws in seq(500, 5000, by = 500)) {
  cat(draws)
  for(i in 1:100) {
    cat(".")
    model <- rstanarm::stan_glm(mpg ~ wt, data = mtcars, iter = draws, refresh = 0)
    ci <- insight::reshape_ci(bayestestR::hdi(model, ci = c(0.8, 0.85, 0.9, 0.95, 0.99)))[2, ]
    data <- rbind(data,
                  data.frame(draws = draws,
                             i = i,
                             CI_80 = ci$CI_low_0.8,
                             CI_85 = ci$CI_low_0.85,
                             CI_90 = ci$CI_low_0.9,
                             CI_95 = ci$CI_low_0.95,
                             CI_99 = ci$CI_low_0.99))
  }
}


data %>% 
  group_by(draws) %>% 
  summarise_all(sd) %>% 
  select(-i) %>% 
  tidyr::pivot_longer(-draws, names_to = "CI", values_to = "SD") %>% 
  ggplot(aes(x = draws, y = SD, color = CI)) +
  geom_line() +
  geom_point() +
  see::theme_modern() +
  scale_color_viridis_d()
```

**Comment by mattansb** [comment]
This is a very simple example. I'm sure across more complex models (more parameters, worse convergence, different likelihoods - esp those that produce non symmetrical posteriors) this would exacerbate the issue.

**Comment by strengejacke** [comment]
Rplot

**Comment by strengejacke** [comment]
Rplot02

**Comment by DominiqueMakowski** [comment]
> This is a very simple example. I'm sure...



🤓👨‍⚕️ It seems obvious to me that yours and Daniel's ardent opposition to 0.95 stems out of 1) an unresolved past trauma that fuels this iconoclasm for @mattansb, and 2) a repressed love of norms that - through reverse mirroring - creates this adoration for a new set of norms for @strengejacke. 
That would be all for today's session you can wake up from the sofa I take only cash 💵

**Comment by DominiqueMakowski** [comment]
Here's an updated figure. Bottom line, the stability argument alone doesn't appear as sufficient to justify our previous threshold. Even in potential cases where this would be exacerbated; yes it's a bit more "stable", but I don't think this amount of precision in estimating the bounds is what the base user that uses the default looks for anyway... and 95 is used for bootstrapped distributions with even less draws than that and nobody bats an eye... And the "at least 10,000 samples" is as another arbitrary number - and I thought you didn't magical numbers 😁 

Anyway, I think we all agree on the fact that we should focus on first and foremost fostering an informed decision (the vignettes and the docstrings are pretty lengthy now about this issue) :)

image


```r
library(dplyr)
library(ggplot2)
library(rstanarm)

data <- data.frame()
for(draws in seq(500, 10000, by = 500)) {
  cat(draws)
  for(i in 1:3000) {
    # cat(".")
    # model <- rstanarm::stan_glm(mpg ~ wt, data = mtcars, iter = draws, refresh = 0)
    # ci <- insight::reshape_ci(bayestestR::hdi(model, ci = c(0.5, 0.7, 0.8, 0.85, 0.9, 0.95, 0.99)))[2, ]
    model <- rnorm(draws)
    ci <- insight::reshape_ci(bayestestR::hdi(model, ci = c(0.5, 0.7, 0.8, 0.85, 0.9, 0.95, 0.99, 0.999)))
    data <- rbind(data,
                  data.frame(draws = draws,
                             i = i,
                             CI_50 = ci$CI_low_0.5,
                             CI_70 = ci$CI_low_0.7,
                             CI_80 = ci$CI_low_0.8,
                             CI_85 = ci$CI_low_0.85,
                             CI_90 = ci$CI_low_0.9,
                             CI_95 = ci$CI_low_0.95,
                             CI_99 = ci$CI_low_0.99,
                             CI_999 = ci$CI_low_0.999))
  }
}


data %>%
  group_by(draws) %>%
  summarise_all(sd) %>%
  select(-i) %>%
  tidyr::pivot_longer(-draws, names_to = "CI", values_to = "SD") %>%
  ggplot(aes(x = draws, y = SD, color = CI)) +
  geom_line() +
  geom_point() +
  see::theme_modern() +
  scale_color_viridis_d()
```

**Comment by DominiqueMakowski** [comment]
I take the silence as a merge greenlight 😋

**Comment by mattansb** [comment]
lol you killed @strengejacke 's startup message?? Harsh...

**Comment by DominiqueMakowski** [comment]
I merely transferred it to its rightful place -> in the vignettes 😁

**Comment by strengejacke** [comment]
Reviewer 2 (@mattansb): "Your model used in the simulation seems to simple to gain evidence for your claims"

@DominiqueMakowski: "We thank you for your comment and replaced the rstanarm model with a less sophisticated way of simulating data rnorm()".

**Comment by DominiqueMakowski** [comment]
hahahaha

**Comment by mattansb** [comment]
😂😂😂😂😂
I do give off a bit of a reviewer 2 vibe, hu?
I just ~like being an a$$~ want to make the paper better!

**Pull request merged**